Add documentation for adding a new track (#279)
exercism/docs: building/tracks/new/select-testing-framework.md

# Select testing framework

All of the exercises are based on providing a failing test suite for people to run and make pass, so we need to pick a testing framework.

> If there are multiple testing frameworks available, we'll need to figure out which one is the best fit.

Reviewer: ```suggestion
If there are multiple testing frameworks available, we will need to figure out which one is the best fit.
```

## Considerations

1. Ease of getting started.
1. Cross-platform support (MacOS, Linux, Windows).

> For people who are going to be doing the exercises, it's important to get them up and running as quickly as possible.

Reviewer: ```suggestion
For people who will be doing the exercises, it's important to get them up and running as quickly as possible.
```

In general we prefer tools in the standard library if they're available, and if not we aim for a third party library that is widely used.
If there is a choice between several common libraries, then consider a few factors:

- Is one of them easier to install?
- Is one of them easier to get help with or search for resources about?

> - Is one of them simpler than the other either in syntax or conceptually?

Reviewer: ```suggestion
- Is one of them simpler than the other either in syntax or concept?
```

Match wording.  "syntactically or conceptually" or "syntax or concept"

> Remember that all Exercism exercises are by nature very small, so there's no need for the complexity that real-world projects sometimes require.

Reviewer: ```suggestion
Remember that all Exercism exercises are by nature very small, so there is no need for the complexity that real-world projects sometimes require.
```

## Document the decisions

As you make these decisions, document them in the track repository's `README.md` for

> contributors and future maintainers. In particular, when someone submits a contribution

Reviewer: ```suggestion
contributors and maintainers. In particular, when someone submits a contribution
```

We do not mention future contributors, and this shows empathy for current maintainers as well.

that goes against one of these decisions, it's very helpful to be able to point to not just
the conclusion, but also a bit about why that decision was made. That makes it easier to
decide whether to just close the issue or pull request from the contributor, or whether

> it's time to re-evaluate the decision based on new information.

Reviewer: ```suggestion
it is time to re-evaluate the decision based on new information.
```

> 

Reviewer: The above paragraph seems to stray from what has been "sentence per line" style.
